File: frontend/src/services/actorService.js
import { apiClient } from "./apiClient";

export class ActorService {

    getActorsByMovieId(movieId) {
        return apiClient.get("/movie/actors/getActorsByMovieId/" + movieId);
    }

    getall() {
        return apiClient.get("/movie/actors/getall");
    }
    
    addActor(actorDto) {
        return apiClient.post("/movie/actors/add", actorDto);
    }
}

File: movieService/src/main/java/com/doffeii/movieService/business/abstracts/ActorService.java
package com.doffeii.movieService.business.abstracts;

import com.doffeii.movieService.entity.Actor;
import com.doffeii.movieService.entity.dto.ActorRequestDto;

import java.util.List;

public interface ActorService {

    List<Actor> getActorsByMovieId(int movieId);

    List<Actor> getall();

    void addActors(ActorRequestDto actorRequestDto);
}
